src/help/help_topic_generators.cpp

```diff
@@ -632,6 +632,10 @@ std::string unit_topic_generator::operator()() const {
 		std::set<t_translation::terrain_code>::const_iterator terrain_it =
 			preferences::encountered_terrains().begin();
 
+
+		// the tuple's contents are: 0 name (t_string), 1 id (t_string), 2 defense (int), 3 movement_cost (int), 4 vision_cost (int), 5 jamming_cost (int), 6 defense_cap (bool)
+		std::vector<std::tuple<t_string, t_string, int, int, int, int, bool>> terrain_moves;
```

Comment:
This is too large to be a tuple. Please create a named struct instead.

Reply:
In addition, a set is a better container here than a vector. A set is automatically sorted, using either `operator<` or a custom comparison function.